SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp

```diff
@@ -149,6 +149,14 @@ bool USpatialPackageMapClient::ResolveEntityActor(AActor* Actor, Worker_EntityId
 	if (!NetGUID.IsValid())
 	{
 		NetGUID = SpatialGuidCache->AssignNewEntityActorNetGUID(Actor, EntityId);
+
+		for (UActorComponent* DynamicComponent : Actor->GetInstanceComponents())
+		{
+			if (DynamicComponent->IsSupportedForNetworking())
+			{
+				TryResolveNewDynamicSubobjectAndGetClassInfo(DynamicComponent);
```

Comment:
I'm happy with this change but would be glad to have some feedback on it - this is the main part where the initial issue is fixed.

Reply:
It seems somewhat loose to do this checking at this point. Wouldn't it be safer to be more explicit about doing this checking when the actor is replicated for the first time?

Reply:
I've found this method as a point where an actor would exist on both the client and the server, and that would be called on both clients and servers. I'll dig around, try to find other customization points.

Reply:
Oh - I assumed this only needed to be called on the auth server. You're saying we need to call this on all workers?

Reply:
Are the server and client guaranteed to iterate these components in the same order? I suspect there will be problems with this approach if one of these components is destroyed before this is called.

Reply:
1. No, I believe they aren't necessarily ordered - would you suggest sorting them? I think this can only cause issues if we have multiple components of the same type though, or am I missing anything?
1. Can you elaborate on the case this could happen? If an actor has a component added to it in the level, and then it's destroyed during, say, server's `BeginPlay` so before this code is called?

Reply:
1. Yes, I was thinking of the case of multiple components of the same type (which isn't unlikely to happen in practice).
2. The component could be destroyed at any point during the first frame before the actor is replicated for the first time. This could be `BeginPlay` on the actor itself or on this attached component or somewhere else. I believe we have seen this happen for static components (where they were destroyed on the server and had to also be destroyed on the client). On that note, what is Unreal native behavior for that with these instance components? If you destroy one on the server, does that get replicated to the clients?

Reply:
1. Yeah totally - a character often has a multitude of skeletal mesh components, for instance. Since this line relates to instance components, these can probably be sorted by names so the order of addition is the same on clients and the servers.
2. Should we destroy the component before replicating it for the first time, the client component would not be destroyed - at least the code seems to suggest that. I'll check Unreal's native behaviour out, but do you happen to have a hint on where to look for static component destruction you've mentioned? Maybe we could leverage this in our case as well.

Reply:
1. Sorting them by name sounds reasonable.
2. I can't seem to find anything in our code to handle that case, so it's likely I misremembered the issue. I think it may have been a case of the static component being destroyed on the client, and then receiving an update from the server for that component leading to crash or invalid behavior. It would still be interesting to see how native Unreal handles the case of the server removing the component prior to replication, both for static components and instanced ones, and whether we can do the same (if not, it may be sufficient to add a known issue).

Reply:
1. Done
2. Neither GDK nor Unreal are destroying the client-side component if one were to destroy the server-side component on BeginPlay so I can say "feature parity" :D